x/stake/keeper/validator.go

```diff
@@ -519,11 +520,15 @@ func (k Keeper) UpdateBondedValidators(
 			oldCliffVal, found := k.GetValidator(ctx, oldCliffValidatorAddr)
 			ensureValidatorFound(found, oldCliffValidatorAddr)
 
-			if bytes.Equal(validatorToBond.OperatorAddr, affectedValidator.OperatorAddr) {
+			affectedValRank := GetValidatorsByPowerIndexKey(affectedValidator, pool)
+			oldCliffValRank := GetValidatorsByPowerIndexKey(oldCliffVal, pool)
 
-				// begin unbonding the old cliff validator iff the affected
-				// validator was newly bonded and has greater power
-				k.beginUnbondingValidator(ctx, oldCliffVal)
+			if bytes.Equal(validatorToBond.OperatorAddr, affectedValidator.OperatorAddr) {
+				if bytes.Compare(affectedValRank, oldCliffValRank) > 0 {
```

Comment:
should these two if statement's want to be combined? (if enters the first but not the second this function won't move on to the else statement below - doesn't sound like the desired functionality) 
________________________________
Think we need to understand the reasons for this solution a bit more... (not sure if worth if for upcoming refactor though).. This fix doesn't really make sense to me